### PR TITLE
Header stream.h moved in tensorflow

### PR DESCRIPTION
**Pull request by EnricoMi**
Signed-off-by: Enrico Minack <github@enrico.minack.dev>

## Checklist before submitting

- [ ] Did you read the [contributor guide](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs?
- [ ] Did you write any tests to validate this change?  
- [ ] Did you update the [CHANGELOG](https://github.com/horovod/horovod/blob/master/CHANGELOG.md), if this change affects users?

## Description

This should fix build failures with TF nightly caused by [2be83013 (tensorflow)](https://github.com/tensorflow/tensorflow/commit/2be830137aca09da21d599e04ba7d7ad0059ed05).

https://github.com/tensorflow/tensorflow/issues/57494
## Review process to land 

1. All tests and other checks must succeed.
2. At least one member of the [technical steering committee](https://github.com/horovod/horovod/blob/master/CONTRIBUTING.md) must review and approve.
3. If any member of the technical steering committee requests changes, they must be addressed.

### CODE REVIEW

**Comment by EnricoMi** [comment]
This isn't the final location of the header file: https://github.com/tensorflow/tensorflow/issues/57494#issuecomment-1239823690

Just testing for now.

File: horovod/tensorflow/mpi_ops.cc [hunk 1]
```diff
@@ -62,7 +62,11 @@ namespace gpu {
 GpuStreamHandle AsGpuStreamValue(Stream* stream);
 } // namespace stream_executor
 } // namespace gpu
+#if TENSORFLOW_VERSION >= 2010000000
+#include "tensorflow/compiler/xla/stream_executor/stream.h"
```
Comment (maxhgerlach) [inline horovod/tensorflow/mpi_ops.cc:66]: This appears to change the build with TF 2.10 (released 14 days ago), not only with recent `tf-nightly`. Is that intentional?

I think current TensorFlow master would present itself as 2.11.0, judging from https://github.com/tensorflow/tensorflow/blob/9c73ada2531dd8ec2efe45a5faf44199aed9524e/tensorflow/tensorflow.bzl#L64

Comment (EnricoMi) [reply horovod/tensorflow/mpi_ops.cc:66]: you are right, thanks for that!

**Comment by EnricoMi** [comment]
@maxhgerlach this is green.

**Review by maxhgerlach — APPROVED** [review]
LGTM, great to have the pipelines pass again! (at least for a while...)

**Comment by EnricoMi** [comment]
After moving to CUDA 11.6, pytorch HEAD is failing.

**Pull request merged**